lib/model/folder.go

```diff
@@ -51,3 +51,7 @@ func (f *folder) scanSubdirsIfHealthy(subDirs []string) error {
 	}
 	return nil
 }
+
+func (f *folder) scanner() *folderScanner {
+	return &f.scan
```

Comment:
I dont think this is worse. It is meant to be a fluent api implementing a getter for the `folderScanner`. The only thing is that I made a mistake using the wrong idioms because I'm not familiar with golang yet.

If I had used your suggestions ...
- it would be confusing with the `func (f *folder) Scan(subdirs []string) error` 
- and the problem with accessing fields directly would not be solved.

I understood @calmh 's hint, to implement to OOP style. With which I fully agree on. This means using methods to access fields to not undermine any future refactoring and to keep stable contracts.